google/cloud/aiplatform/matching_engine/matching_engine_index_endpoint.py

```diff
@@ -1207,10 +1208,12 @@ def deploy_index(
                 auth_config_audiences and auth_config_allowed_issuers must be passed together.
             request_metadata (Sequence[Tuple[str, str]]):
                 Optional. Strings which should be sent along with the request as metadata.
-
+            sync (bool):
+                Whether to execute this method synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
```

Comment:
I don't think the sync portion of this is true unless you add the `@base.optional_sync()` annotation to this method.